core/storage/couchStorage.go

```diff
@@ -0,0 +1,1801 @@
+	result.LastUpdate = time.Now()
+
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to mark object as active. Error: %s.", err)}
+	}
+	return nil
+}
+
+// UpdateObjectStatus updates object's status
+func (store *CouchStorage) UpdateObjectStatus(orgID string, objectType string, objectID string, status string) common.SyncServiceError {
+	id := createObjectCollectionID(orgID, objectType, objectID)
+	result := couchObject{}
+	if err := store.getOne(id, &result); err != nil {
+		return &Error{fmt.Sprintf("Failed to retrieve object. Error: %s.", err)}
+	}
+	result.Status = status
+	result.LastUpdate = time.Now()
+
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to update object's status. Error: %s.", err)}
+	}
+	return nil
+}
+
+// RetrieveUpdatedObjects returns the list of all the edge updated objects that are not marked as consumed or received
+// If received is true, return objects marked as received
+func (store *CouchStorage) RetrieveUpdatedObjects(orgID string, objectType string, received bool) ([]common.MetaData, common.SyncServiceError) {
+	result := []couchObject{}
+	var query interface{}
+	if received {
+		query = map[string]interface{}{
+			"selector": map[string]interface{}{
+				"$or": []map[string]interface{}{
+					map[string]interface{}{"status": common.CompletelyReceived},
+					map[string]interface{}{"status": common.ObjReceived},
+					map[string]interface{}{"status": common.ObjDeleted}},
+				"metadata.destinationOrgID": orgID, "metadata.objectType": objectType}}
+	} else {
+		query = map[string]interface{}{
+			"selector": map[string]interface{}{
+				"$or": []map[string]interface{}{
+					map[string]interface{}{"status": common.CompletelyReceived},
+					map[string]interface{}{"status": common.ObjDeleted}},
+				"metadata.destinationOrgID": orgID, "metadata.objectType": objectType}}
+	}
+
+	if err := store.findAll(query, &result); err != nil {
+		return nil, &Error{fmt.Sprintf("Failed to fetch the objects. Error: %s.", err)}
+	}
+
+	metaDatas := make([]common.MetaData, len(result))
+	for i, r := range result {
+		metaDatas[i] = r.MetaData
+	}
+	return metaDatas, nil
+}
+
+// RetrieveObjects returns the list of all the objects that need to be sent to the destination.
+// Adds the new destination to the destinations lists of the relevant objects.
+func (store *CouchStorage) RetrieveObjects(orgID string, destType string, destID string, resend int) ([]common.MetaData, common.SyncServiceError) {
+	result := []couchObject{}
+	query := map[string]interface{}{
+		"selector": map[string]interface{}{"metadata.destinationOrgID": orgID,
+			"$or": []map[string]interface{}{
+				map[string]interface{}{"status": common.ReadyToSend},
+				map[string]interface{}{"status": common.NotReadyToSend},
+			}}}
+
+	if err := store.findAll(query, &result); err != nil {
+		return nil, &Error{fmt.Sprintf("Failed to fetch the objects. Error: %s.", err)}
```

Comment:
Like the comment above, return `nil nil` if the err is notFound (like the implementation in mongoStorage)